config/options_test.go

```diff
@@ -505,6 +505,41 @@ func TestHTTPRedirectAddressStripQuotes(t *testing.T) {
 	assert.Equal(t, ":80", o.HTTPRedirectAddr)
 }
 
+func TestCertificatesArrayParsing(t *testing.T) {
+	t.Parallel()
+
+	testCertFileRef := "./testdata/example-cert.pem"
+	testKeyFileRef := "./testdata/example-key.pem"
+	testCertFile, _ := ioutil.ReadFile(testCertFileRef)
+	testKeyFile, _ := ioutil.ReadFile(testKeyFileRef)
+	testCertAsBase64 := base64.StdEncoding.EncodeToString(testCertFile)
+	testKeyAsBase64 := base64.StdEncoding.EncodeToString(testKeyFile)
+
+	tests := []struct {
+		name             string
+		certificateFiles []certificateFilePair
+		wantErr          bool
+	}{
+		{ "Handles base64 string as params", []certificateFilePair{{ KeyFile: testKeyAsBase64, CertFile: testCertAsBase64 }}, false },
+		{ "Handles file reference as params", []certificateFilePair{{ KeyFile: testKeyFileRef, CertFile: testCertFileRef }}, false },
+		{ "Returns an error otherwise", []certificateFilePair{{ KeyFile: "abc", CertFile: "abc" }}, true },
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
You need `tt := tt` before `t.Run`, so it won't capture wrong `tt` variable in for loop.

Reply:
Will add, but could you please elaborate? I assumed that Go would introduce a new scope when the test func was declared. As such there should be no scope confusion about `tt`?

Thanks!

Reply:
I invite you to read https://blog.golang.org/subtests, read `Run a group of tests in parallel` section.

Tests run without parallel is safe to not capture the range variable.

Reply:
Thank you! I made the change, but this is a great starting point for me understanding why :) Appreciate your patience!

Reply:
> Thank you! I made the change, but this is a great starting point for me understanding why :) Appreciate your patience!

Would you please formatting files with `gofmt -s` 😄